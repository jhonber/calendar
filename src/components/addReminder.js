import React from 'react'
import { Form } from 'react-bootstrap'
import { HuePicker } from 'react-color'
import DatePicker from 'react-datepicker'
import 'react-datepicker/dist/react-datepicker.css'

export default class AddReminder extends React.Component {
  render () {
    return (
      <Form>
        <Form.Group controlId='formBasicEmail'>
          <HuePicker
            width='100%'
            color={this.props.state.color}
            onChangeComplete={this.props.handleColor}
          />
          <Form.Control
            onChange={(e) => this.props.handleText(e)}
            type='text'
            placeholder='Text'
          />
          <Form.Control
            onChange={(e) => this.props.handleCity(e)}
            type='text'
            placeholder='City'
          />
          <DatePicker
            selected={this.props.state.date}
            onChange={date => this.props.handleDate(date)}
          />

          <DatePicker
            selected={this.props.state.time}
            onChange={time => this.props.handleTime(time)}
            showTimeSelect
            showTimeSelectOnly
            timeFormat='HH:mm'
            timeIntervals={60}
            timeCaption='time'
            dateFormat='HH:mm'
          />
        </Form.Group>
      </Form>
    )
  }
}
