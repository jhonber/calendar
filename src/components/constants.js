export const MaxRemindersToList = 2

export const daysOfWeek = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday'
]

export const monthsOfYear = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
]
