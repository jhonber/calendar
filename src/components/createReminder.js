import React from 'react'
import { Button, Form } from 'react-bootstrap'
import { HuePicker } from 'react-color'
import DatePicker from 'react-datepicker'
import 'react-datepicker/dist/react-datepicker.css'

export default class CreateReminder extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      color: '#FF6900',
      text: '',
      city: '',
      date: new Date(this.props.date),
      time: new Date().getTime(),
      validated: false
    }

    this.handleColor = this.handleColor.bind(this)
    this.handleText = this.handleText.bind(this)
    this.handleCity = this.handleCity.bind(this)
    this.handleDate = this.handleDate.bind(this)
    this.handleTime = this.handleTime.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleColor (color, event) {
    this.setState({
      color: color.hex
    })
  }

  handleText (e) {
    this.setState({
      text: e.target.value
    })
  }

  handleCity (e) {
    this.setState({
      city: e.target.value
    })
  }

  handleTime (time) {
    this.setState({
      time: time.getTime()
    })
  }

  handleDate (date) {
    console.log('DATE: ', date)
    this.setState({
      date: date
    }, () => {
      console.log('UPDATED date')
      console.log(this.state.date)
    })
  }

  handleSubmit (e) {
    const form = document.getElementById('createFormId')
    if (form.checkValidity() === false) {
      e.preventDefault()
      e.stopPropagation()
    } else {
      this.props.handleSubmitCreateReminder(this.state)
    }

    this.setState({
      validated: true
    })
  }

  render () {
    return (
      <Form
        noValidate
        validated={this.state.validated}
        onSubmit={this.handleSubmit}
        id='createFormId'
      >
        <Form.Group controlId='formBasicEmail' className='createForm'>
          <HuePicker
            width='100%'
            color={this.state.color}
            onChangeComplete={this.handleColor}
          />
          <Form.Control
            onChange={(e) => this.handleText(e)}
            type='text'
            placeholder='Text'
            maxLength={30}
            required
          />
          <Form.Control.Feedback type='invalid'>
              Please enter a text (30 max)
          </Form.Control.Feedback>
          <Form.Control
            onChange={(e) => this.handleCity(e)}
            type='text'
            placeholder='City'
            required
          />
          <Form.Control.Feedback type='invalid'>
              Please enter a city
          </Form.Control.Feedback>
          <DatePicker
            selected={this.state.date}
            onChange={date => this.handleDate(date)}
          />
          <DatePicker
            selected={this.state.time}
            onChange={time => this.handleTime(time)}
            showTimeSelect
            showTimeSelectOnly
            timeFormat='HH:mm'
            timeIntervals={60}
            timeCaption='time'
            dateFormat='HH:mm'
          />
        </Form.Group>
        <Button
          type='submit'
          className='actionButton'
        >
          Create
        </Button>
      </Form>
    )
  }
}
