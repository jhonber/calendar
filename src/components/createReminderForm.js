import React from 'react'
import { Button, Form } from 'react-bootstrap'
import { HuePicker } from 'react-color'
import DatePicker from 'react-datepicker'
import 'react-datepicker/dist/react-datepicker.css'

export default class CreateReminderForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      color: '#FF6900',
      text: '',
      city: '',
      date: new Date(this.props.date),
      time: new Date(this.props.date).getTime(),
      validated: false
    }

    this.handleColor = this.handleColor.bind(this)
    this.handleText = this.handleText.bind(this)
    this.handleCity = this.handleCity.bind(this)
    this.handleDate = this.handleDate.bind(this)
    this.handleTime = this.handleTime.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleColor (color, event) {
    this.setState({
      color: color.hex
    })
  }

  handleText (e) {
    this.setState({
      text: e.target.value
    })
  }

  handleCity (e) {
    this.setState({
      city: e.target.value
    })
  }

  handleTime (time) {
    this.setState({
      time: time.getTime()
    })
  }

  handleDate (date) {
    console.log('DATE: ', date)
    this.setState({
      date: date
    })
  }

  handleSubmit (event) {
    event.preventDefault()
    event.stopPropagation()
    const form = document.getElementById('createFormId')

    if (form.checkValidity() !== false) {
      this.props.handleSubmit(this.state)
      this.props.toggleModal()
    }

    this.setState({
      validated: true
    })
  }

  render () {
    return (
      <div className='createForm'>
        <Form
          noValidate
          validated={this.state.validated}
          onSubmit={this.handleSubmit}
          id='createFormId'
        >
          <Form.Group controlId='formBasicEmail' className='createForm'>
            <div className='inputField'>
              <HuePicker
                width='100%'
                color={this.state.color}
                onChangeComplete={this.handleColor}
              />
            </div>
            <div className='inputField'>
              <Form.Control
                onChange={(e) => this.handleText(e)}
                type='text'
                placeholder='Text'
                maxLength={30}
                required
              />
              <Form.Control.Feedback type='invalid'>
                Please enter a text (30 max)
              </Form.Control.Feedback>
            </div>
            <div className='inputField'>
              <Form.Control
                onChange={(e) => this.handleCity(e)}
                type='text'
                placeholder='City'
                required
              />
              <Form.Control.Feedback type='invalid'>
                  Please enter a city
              </Form.Control.Feedback>
            </div>
            <div className='inputField'>
              <DatePicker
                selected={this.state.date}
                onChange={date => this.handleDate(date)}
              />
            </div>
            <div className='inputField'>
              <DatePicker
                selected={this.state.time}
                onChange={time => this.handleTime(time)}
                showTimeSelect
                showTimeSelectOnly
                timeFormat='HH:mm'
                timeIntervals={60}
                timeCaption='time'
                dateFormat='HH:mm'
              />
            </div>
          </Form.Group>
          <Button
            type='submit'
            className='actionButton'
          >
            Create
          </Button>
        </Form>
      </div>
    )
  }
}
