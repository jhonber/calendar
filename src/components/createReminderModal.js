import React from 'react'
import { connect } from 'react-redux'
import { addReminder } from '../redux/actions/reminders'
import '../App.css'

import moment from 'moment'
import Modal from './modal'
import ReminderForm from './reminderForm'

export class CreateReminderModal extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}

    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit (data) {
    this.props.addReminder({
      color: data.color,
      text: data.text,
      city: data.city,
      date: moment(data.date),
      time: data.time
    })
  }

  render () {
    const body = <ReminderForm
      date={this.props.date}
      toggleModal={this.props.toggleModal}
      handleSubmit={this.handleSubmit}
      labelButton='Create'
    />

    return (
      <div>
        <Modal
          visible={this.props.visible}
          body={body}
          okButton
          okButtonText='Add'
          closeButton
          handleOkButton={() => this.handleSubmit()}
          handleToggleModal={this.props.toggleModal}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  addReminder: (reminder) => dispatch(addReminder(reminder))
})

export default connect(null, mapDispatchToProps)(CreateReminderModal)
