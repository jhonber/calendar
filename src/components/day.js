import React from 'react'
import { connect } from 'react-redux'
import { getRequest } from './apiUtils/utils.js'
import { delReminder } from '../redux/actions/reminders'
import CONFIG from '../config.json'
import '../App.css'

import Modal from './modal'
import ShowReminder from './showReminder'
import EditReminder from './editReminder'

const ENV = CONFIG.env
const MAX_ITEMS_TO_LIST = 2

class Day extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      body: null
    }
    this.handleClickOnItem = this.handleClickOnItem.bind(this)
    this.handleClickEditReminder = this.handleClickEditReminder.bind(this)
    this.handleClickRemoveReminder = this.handleClickRemoveReminder.bind(this)
  }

  handleToggleModal () {
    this.setState({
      showModal: !this.state.showModal
    })
  }

  renderShowReminder (data) {
    const body = <ShowReminder
      handleClickEditReminder={this.handleClickEditReminder}
      handleClickRemoveReminder={this.handleClickRemoveReminder}
      data={data}
    />

    this.setState({
      showModal: true,
      body: body
    })
  }

  handleClickOnItem (data) {
    const query = CONFIG.weather + data.city
    const url = `${CONFIG[ENV].urlBase}${query}`
    const self = this

    getRequest(url, data.city).then((res) => {
      if (res && Object.prototype.hasOwnProperty.call(res, 'weather')) {
        self.renderShowReminder({
          ...data,
          weather: res.weather[0].main
        })
      } else {
        self.renderShowReminder({
          ...data
        })
      }
    }).catch((err) => {
      console.log('Error: ', err)
      self.renderShowReminder({
        ...data
      })
    })
  }

  handleClickCreateReminder (event) {
    if (!this.props.disable && event.target.id === 'day-square') {
      this.props.handleClickCreateReminder(this.props.date)
    }
  }

  handleClickEditReminder (data) {
    const body = <EditReminder
      reminder={data}
    />

    this.setState({
      body: body
    })
  }

  handleClickRemoveReminder (id) {
    if (window.confirm('Remove reminder?')) {
      this.props.dispatch(delReminder(
        { id: id }
      ))
    }
  }

  render () {
    const itemsToList = this.props.items.slice(0, MAX_ITEMS_TO_LIST)
    const rest = this.props.items.length - itemsToList.length
    const overflow = rest > 0
      ? <p>{rest} more</p>
      : null
    let classesLabel = 'Day-label' + (this.props.disable ? ' Disable-day' : '')
    classesLabel = this.props.holiday && !this.props.disable
      ? classesLabel + ' holiday-label'
      : classesLabel

    const classesMain = this.props.holiday
      ? 'Day holiday'
      : 'Day'

    return (
      <div
        className={classesMain}
        id='day-square'
        onClick={(event) => this.handleClickCreateReminder(event)}
      >
        <div className={classesLabel}>
          {this.props.label}
        </div>
        <div className='Items'>
          <ul>
            {itemsToList.map((item, cnt) => {
              return (
                <div key={item.id + cnt} className='Row-style'>
                  <li
                    style={{ backgroundColor: item.color }}
                    key={item.id}
                    onClick={() => this.handleClickOnItem(item)}
                  >
                    {item.text}
                  </li>
                </div>
              )
            })}
            {overflow}
          </ul>
        </div>
        <Modal
          visible={this.state.showModal}
          body={this.state.body}
          closeButton
          handleToggleModal={() => this.handleToggleModal()}
        />
      </div>
    )
  }
}

export default connect()(Day)
