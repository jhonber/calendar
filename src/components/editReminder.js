import React from 'react'
import { connect } from 'react-redux'
import { editReminder } from '../redux/actions/reminders'
import '../App.css'

import moment from 'moment'
import ReminderForm from './reminderForm'

class EditReminder extends React.Component {
  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit (data) {
    this.props.dispatch(editReminder(data.id, {
      color: data.color,
      text: data.text,
      city: data.city,
      date: moment(data.date),
      time: data.time
    }))
  }

  render () {
    return (
      <ReminderForm
        date={this.props.date}
        reminder={this.props.reminder}
        handleSubmit={this.handleSubmit}
      />
    )
  }
}

export default connect()(EditReminder)
