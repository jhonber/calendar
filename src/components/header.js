import React from 'react'
import '../App.css'

export default class Header extends React.Component {
  render () {
    return (
      <div className={this.props.styleName}>
        {this.props.label}
      </div>
    )
  }
}
