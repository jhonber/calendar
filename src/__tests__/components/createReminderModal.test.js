import React from 'react'
import { shallow } from 'enzyme'
import { CreateReminderModal } from '../../components/createReminderModal'
import toJSON from 'enzyme-to-json'
import moment from 'moment'

/*eslint-disable */
describe('create reminder modal', () => {
  it('should render create reminder modal correctly', () => {
    const props = {
      visible: true,
      toggleModal: jest.fn(),
      date: '2019-11-10'
    }

    const wrapper = shallow(<CreateReminderModal {...props} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should call addReminder with correct data', () => {
    const reminder = {
      text: 'My cool reminder',
      city: 'Pereira',
      color: '#cddc39',
      date: '2019-11-10',
      time: '20:00'
    }

    const props = {
      visible: true,
      toggleModal: jest.fn(),
      date: '2019-11-10',
      reminder: reminder,
      addReminder: jest.fn()
    }

    const expectedData = {
      text: reminder.text,
      city: reminder.city,
      color: reminder.color,
      date: moment(reminder.date),
      time: reminder.time
    }

    const wrapper = shallow(<CreateReminderModal {...props} />)
    wrapper.props().children.props.body.props.handleSubmit(reminder)
    expect(props.addReminder.mock.calls[0][0]).toEqual(expectedData)
  })
})