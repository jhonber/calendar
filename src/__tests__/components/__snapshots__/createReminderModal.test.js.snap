// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create reminder modal should render create reminder modal correctly 1`] = `
<div>
  <MyModal
    body={
      <ReminderForm
        date="2019-11-10"
        handleSubmit={[Function]}
        labelButton="Create"
        toggleModal={[MockFunction]}
      />
    }
    closeButton={true}
    handleToggleModal={[MockFunction]}
    okButton={true}
    okButtonText="Add"
    visible={true}
  />
</div>
`;
